plugins/hls-code-range-plugin/src/Ide/Plugin/CodeRange.hs

```diff
@@ -154,7 +116,7 @@ getSelectionRanges file positions = do
              in fromMaybe defaultSelectionRange . findPosition pos $ codeRange
 
     -- 'positionMapping' should be applied to the output ranges before returning them
-    maybeToExceptT SelectionRangeOutputPositionMappingFailure . MaybeT . pure $
+    maybeToExceptT (PluginDependencyFailed "toCurrentSelectionRange") . MaybeT . pure $
```

Comment:
wrong error kind?

Reply:
We use PluginDependencyFailed for all failed dependency besides rule failures. In this case it's the function toCurrentSelectionRange (which relies on PositionMapping)

Reply:
I think there should be a distinction from your plugin didn't work because the user's code is in an invalid state, and your code didn't work because there is a bug in it.

Reply:
Okay, but that's a third thing! I think by that you mean the code buffer is in a weird state, right? So that could manifest as:
- Some rule that you need fails (`BadDependency` or whatever)
- The rules all succeed but we can't process the result for some reason. Position mapping failing seems like a good example.

I totally agree the latter is a valid category, I just think it's weird to call it "DependencyFailed"! Maybe something like:
- `InternalError` -> a bug in our code, you should file an issue
- `DependencyFailed` -> a rule dependency failed, a "normal" thing that happens often but means we can't process the request
- `HandlerFailed` -> the handler failed, but in a "normal" way that doesn't indicate a bug

Reply:
what about PluginInvalidUserState ? That at least clarifies in the name that the failure can only be about the user's state vs PluginHandlerFailed being more generic?